pkg/controller/jobs/appwrapper/appwrapper_controller.go

```diff
@@ -120,20 +121,35 @@ func (aw *AppWrapper) GVK() schema.GroupVersionKind {
 }
 
 func (aw *AppWrapper) PodSets() ([]kueue.PodSet, error) {
-	podSets, err := awutils.GetPodSets((*awv1beta2.AppWrapper)(aw))
+	podSpecTemplates, awPodSets, err := awutils.GetComponentPodSpecs((*awv1beta2.AppWrapper)(aw))
 	if err != nil {
-		ctrl.Log.Error(err, "Error returned from awutils.GetPodSets", "appwrapper", aw)
+		ctrl.Log.Error(err, "Error returned from awutils.GetComponentPodSpecs", "appwrapper", aw)
 		return nil, err
 	}
-	for idx := range podSets {
-		podSets[idx].TopologyRequest = jobframework.PodSetTopologyRequest(&podSets[idx].Template.ObjectMeta, nil, nil, nil)
+	podSets := []kueue.PodSet{}
```

Comment:
```suggestion
	podSets := make([]kueue.PodSet, len(podSpecTemplates))
```

There is no need to prepare an empty slice. In Go, an empty slice indicates a special meaning rather than initializing.

Reply:
Changed to use make to pre-allocate a slice of the right length.